### PR TITLE
[Backup File] fix #295310: add advanced preference option of disabling backup

### PR DESCRIPTION
**Pull request by Howard-C**
Resolves: https://musescore.org/node/295310.

### CODE REVIEW

**Comment by Howard-C** [comment]
See https://github.com/Howard-C/MuseScore/tree/backup for this commit. It also adds more commits related to backup.

**Comment by Marr11317** [comment]
Doesn't contain the other improvements?

**Comment by Howard-C** [comment]
> Doesn't contain the other improvements?

We determined to have an alternate solution. Will be done later.

File: libmscore/score.h [hunk 1]
```diff
@@ -1283,7 +1283,7 @@ class MasterScore : public Score {
       bool isSavable() const;
       void setTempomap(TempoMap* tm);
 
-      bool saveFile();
+      bool saveFile(bool generateBackup = true);
```
Comment (Marr11317) [inline libmscore/score.h]: Files saved from within the plugin api should not genererate a backup, no?
So [this](https://github.com/Howard-C/MuseScore/blob/master/mscore/plugin/api/qmlpluginapi.cpp#L134) should add a `false` somewhere?
Maybe that should be on a musescore issue...

Comment (Howard-C) [reply libmscore/score.h]: If generateBackup is true the code isn't anything different from the way it is now. So anything else is a different issue.

Comment (Marr11317) [reply libmscore/score.h]: my error, it already doesn't generate any backup

Comment (Jojo-Schmitz) [reply libmscore/score.h]: which I'd regard a bug

Comment (Marr11317) [reply libmscore/score.h]: @Jojo-Schmitz you think the plugin framework should save a backup of what it writes?
In its current implemenentation, it "saves a copy" of the score.

Comment (Jojo-Schmitz) [reply libmscore/score.h]: If a plugin modifies an existing score, it should create a backup. If it does the equivalent of "Save as" it should not

**Pull request merged**